docs/cow-protocol/tutorials/arbitrate/solver/driver.md

```diff
@@ -128,6 +128,45 @@ sequenceDiagram
+
+The entry point to the router contract ([IFlashLoanRouter](../../../reference/contracts/periphery/flash-loans.md#iflashloanrouter-contract)) is the function `flashLoanAndSettle`.
+It takes a list of loans with the following entries for each loan:
+
+- The loaned amount and ERC-20 token.
+- The flash-loan lender (e.g., Balancer, Aave, Maker, ...).
+- The _borrower_, which is an adapter contract that makes the specific lender implementation compatible with the router.
+
+It also takes the exact call data for a call to `settle`.
+The flash-loan router is a solver for CoW Protocol and calls `settle` directly once the flash loans have been obtained.
+Only CoW-Protocol solvers can call this function.
+
+Tokens and lenders are external contracts, while the router and borrowers have a dedicated contract implemented by CoW Protocol.
+
+The borrowers are the contracts that are called back by the lender once the flash loan is initiated; they are the contracts that receive the flash-loan proceeds and that are eventually responsible to repay the loan.
+
+The only way to move funds out of a borrower is through an ERC-20 approval transaction from the spender.
+Approvals can be set by calling the [approve](../../../reference/contracts/periphery/flash-loans.md#approve) function on the borrower contract ([IBorrower](../../../reference/contracts/periphery/flash-loans.md#iborrower-contract)) from the context of a settlement.
+For safe operations, like an approval for the settlement contract to spend the funds of the borrower, it's enough to set the approval once for an unlimited amount and reuse the same approval in future settlements.
+
+At the start of the settlement, it's expected that the loaned funds are transferred from the borrowers to where they are needed. For example, this can be the settlement contract itself, or the address of a user who wants to use the loan to retrieve the collateral needed to avoid liquidations.
+
+In general, solvers have full flexibility in deciding how loaned funds are allocated.
+
+The settlement is also responsible for repaying the flash loans.
+The specific repayment mechanism depends on the lender, but a common process is having the settlement contract send back the borrowed funds to the borrower and set an approval to the lender for spending the funds of the borrower: then the lender is going to pull back the funds with an ERC-20 `transferFrom` after the settlement is terminated.
+Inability to pay for a flash loan will most likely be met by a reverting transaction.
```

Comment:
There is no doubt about that. Even if the adapter contracts are buggy at the very latest the actual lender contracts will enforce this.
```suggestion
Inability to pay for a flash loan will result in a reverting transaction.
```